File: src/main/kotlin/com/example/cardgenerator/service/CardService.kt
package com.example.cardgenerator.service

import com.example.cardgenerator.model.Card
import com.example.cardgenerator.repository.CardRepository
import org.springframework.stereotype.Service

@Service
class CardService(private val cardRepository: CardRepository) {

    fun getAllCards(): List<Card> = cardRepository.findAll()
    
    fun getCardById(id: Long): Card = cardRepository.findById(id)
        .orElseThrow { NoSuchElementException("Card not found with id: $id") }
    
    fun createCard(card: Card): Card = cardRepository.save(card)
    
    fun updateCard(id: Long, card: Card): Card {
        return if (cardRepository.existsById(id)) {
            cardRepository.save(card.copy(id = id))
        } else {
            throw NoSuchElementException("Card not found with id: $id")
        }
    }
    
    fun deleteCard(id: Long) {
        if (cardRepository.existsById(id)) {
            cardRepository.deleteById(id)
        } else {
            throw NoSuchElementException("Card not found with id: $id")
        }
    }
} 

File: frontend/src/services/CardService.js
import axios from 'axios';

const API_URL = '/api/cards';

class CardService {
  getAllCards() {
    return axios.get(API_URL);
  }

  getCardById(id) {
    return axios.get(`${API_URL}/${id}`);
  }

  createCard(card) {
    return axios.post(API_URL, card);
  }

  updateCard(id, card) {
    return axios.put(`${API_URL}/${id}`, card);
  }

  deleteCard(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
}

export default new CardService(); 